### ISSUE TITLE
Out of memory error on T4 instance

### ISSUE THREAD

**Issue by vinayhpandya** [huggingface/smol-course#141]
Currently getting out of memory error for training a model (using TF32) for DPO

**Comment by Ronit33** [comment]
hi did you try lowering the `per_device_train_batch_size` and `gradient_accumulation_steps` in `DPOConfig`? this will reduce your  effective batch size, also try changing the frequency of your logging steps from 1 to a larger number
its working on my local gpu with this.
if this does not work either then try lowering the: `max_length`

**Comment by vinayhpandya** [comment]
Yes I did, I will try lowering the max_length and try running the notebook again. Do we also need to apply any kind of chat template before finetuning?

**Comment by vinayhpandya** [comment]
It worked I just had to reduce `max_tokens` parameter in `DPOConfig` although the output quality is still pretty bad. I will experiment with PEFT later

**Comment by vinayhpandya** [comment]
Closing this for now
